mmlearn/tasks/linear_probing.py

```diff
@@ -0,0 +1,454 @@
+                )
+                for k in self.top_k_list
+            }
+
+            # Additional metrics for multiclass classification
+            additional_metrics = {
+                "precision": Precision(
+                    task=task, num_classes=num_classes, average="macro"
+                ),
+                "recall": Recall(task=task, num_classes=num_classes, average="macro"),
+                "f1_score": F1Score(
+                    task=task, num_classes=num_classes, average="macro"
+                ),
+                "auc": AUROC(
+                    task=task, num_classes=num_classes, average="macro"
+                ),  # AUROC for multiclass
+            }
+
+        elif task == "multilabel":
+            # Accuracy and other metrics for multilabel classification
+            accuracy_metrics = {"accuracy": Accuracy(task=task, num_labels=num_classes)}
+
+            # Additional metrics for multilabel classification
+            additional_metrics = {
+                "precision": Precision(
+                    task=task, num_labels=num_classes, average="macro"
+                ),
+                "recall": Recall(task=task, num_labels=num_classes, average="macro"),
+                "f1_score": F1Score(task=task, num_labels=num_classes, average="macro"),
+                "auc": AUROC(task=task, num_labels=num_classes),  # AUC for multilabel
+            }
+
+        else:  # binary
+            # Accuracy and other metrics for binary classification
+            accuracy_metrics = {"accuracy": Accuracy(task=task)}
+
+            # Additional metrics for binary classification
+            additional_metrics = {
+                "precision": Precision(task=task),
+                "recall": Recall(task=task),
+                "f1_score": F1Score(task=task),
+                "auc": AUROC(task=task),  # AUROC for binary classification
+            }
+
+        # combine all metrics
+        metrics = MetricCollection({**accuracy_metrics, **additional_metrics})
+        self.train_metrics = metrics.clone(prefix="train/")
+        self.valid_metrics = metrics.clone(prefix="val/")
+
+        self.optimizer = optimizer
+        self.lr_scheduler = lr_scheduler
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Perform a forward pass through the encoder and linear classifier.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor.
+
+        Returns
+        -------
+        torch.Tensor
+            The logits predicted by the classifier.
+        """
+        with torch.no_grad() if self.freeze_encoder else nullcontext():
+            x = self.encoder(x)
+        return self.classifier(x[0])
+
+    def _get_logits_and_labels(
+        self, batch: Dict[str, torch.Tensor]
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Return the logits and labels for a batch of data."""
+        x: torch.Tensor = batch
```

Comment:
The input data corresponding to `modality` is not being extracted from the batch.